listenbrainz/replay_user.py

```diff
@@ -0,0 +1,93 @@
+""" This script should be used to replay user listens to fix bad data.
+"""
+import json
+import uuid
+
+from datetime import datetime
+from flask import current_app
+from listenbrainz.utils import get_measurement_name, quote, convert_to_unix_timestamp
+from listenbrainz.webserver import create_app
+from listenbrainz.webserver.influx_connection import init_influx_connection
+from listenbrainz.listenstore.influx_listenstore import DUMP_CHUNK_SIZE
+
+
+class UserReplayer:
+    def __init__(self, user_name):
+        self.user_name = user_name
+        self.max_time = datetime.now()
+        self.app = create_app()
+
+
+    def filter_function(self, row):
+        return row
+
+
+    def convert_to_influx_insert_format(self, row, measurement):
+        data = {
+            'measurement': measurement,
+            'time': convert_to_unix_timestamp(row['time']),
+        }
+
+        if 'dedup_tag' in row:
```

Comment:
Hmm. This is the innermost loop, meaning that this will be executed gazillions of times and ought to be tight. I wonder if it would be faster to copy all the fields over (using a built-in optimized function and then overwrite the time/dedup tags?